### PR TITLE
Rename RunnableMap to RunnableParallel

### PR DESCRIPTION
**Pull request by nfcampos**
- keep alias for RunnableMap
- update docs to use RunnableParallel and RunnablePassthrough.assign

### CODE REVIEW

File: libs/langchain/langchain/schema/runnable/base.py [hunk 1]
```diff
@@ -1428,16 +1428,28 @@ class RunnableMap(RunnableSerializable[Input, Dict[str, Any]]):
...
         self,
-        steps: Mapping[
-            str,
-            Union[
-                Runnable[Input, Any],
-                Callable[[Input], Any],
-                Mapping[str, Union[Runnable[Input, Any], Callable[[Input], Any]]],
-            ],
+        /,
+        steps: Optional[
```
Comment (eyurtsev) [inline libs/langchain/langchain/schema/runnable/base.py:1432]: Should we make a breaking change and remove `steps`?

**Review by eyurtsev — APPROVED** [review]

File: libs/langchain/langchain/schema/runnable/base.py [hunk 2]
```diff
@@ -1428,16 +1428,27 @@ class RunnableMap(RunnableSerializable[Input, Dict[str, Any]]):
...
-                Mapping[str, Union[Runnable[Input, Any], Callable[[Input], Any]]],
-            ],
+        __steps: Optional[
+            Mapping[
+                str,
+                Union[
+                    Runnable[Input, Any],
+                    Callable[[Input], Any],
+                    Mapping[str, Union[Runnable[Input, Any], Callable[[Input], Any]]],
+                ],
```
Comment (baskaryan) [inline libs/langchain/langchain/schema/runnable/base.py:1438]: for readability maybe worth aliasing

File: libs/langchain/langchain/schema/runnable/base.py [hunk 3]
```diff
@@ -1428,16 +1428,27 @@ class RunnableMap(RunnableSerializable[Input, Dict[str, Any]]):
...
+            ]
+        ] = None,
+        **kwargs: Union[
```
Comment (baskaryan) [inline libs/langchain/langchain/schema/runnable/base.py:1441]: doesn't **kwargs have to be mapping?

File: libs/langchain/langchain/schema/runnable/base.py [hunk 4]
```diff
@@ -1725,6 +1736,9 @@ async def input_aiter() -> AsyncIterator[Input]:
             yield chunk
 
 
+RunnableMap = RunnableParallel
```
Comment (baskaryan) [inline libs/langchain/langchain/schema/runnable/base.py:1739]: can we add comment

File: libs/langchain/langchain/schema/runnable/base.py (hunk 3 above)
Comment (nfcampos) [reply libs/langchain/langchain/schema/runnable/base.py:1441]: Strangely the annotation you attach to kwargs is actually the annotation for the value of the mapping. Same with *args

File: libs/langchain/langchain/schema/runnable/base.py (hunk 2 above)
Comment (nfcampos) [reply libs/langchain/langchain/schema/runnable/base.py:1438]: I dont think aliases work with type variables in python correctly

**Pull request merged**